logger/options.go

```diff
@@ -40,43 +39,66 @@ type Options struct {
 	Timestamp     time.Time
 }
 
-func (e InvalidOptionMessage) String() string {
-	return fmt.Sprintf("Options.%s: %s", e.Option, e.Message)
+type fieldIssue struct {
+	field string
+	prob  string
 }
 
-func validateOptionLength(option string, value string, problems *[]string) {
-	if len(value) > 32 {
-		*problems = append(*problems, InvalidOptionMessage{option, "length must be less than 32"}.String())
+type optionsError struct {
+	issues []fieldIssue
+}
+
+func (e *optionsError) Error() string {
+	errorString := ""
+	for i := 0; i < len(e.issues); i++ {
+		errorString += fmt.Sprintf("%s: %s \n", e.issues[i].field, e.issues[i].prob)
+	}
```

Comment:
if you define a `.String()` method on `fieldIssue`, you won't need to iterate and build up a string:

```golang
func (fi *fieldIssue) String() string {
    return fmt.Sprintf("Options.%s: %s", fi.field, fi.prob)
}

func (e *optionsError) Error() string {
    errors.New(strings.Join(e.issues, "\n"))
}
```

Reply:
I don't think you can. `strings.Join` expects a slice of strings: https://play.golang.org/p/eTcUKj5W7LN

Reply:
ah you're right, no implicit coercion